tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,1639 @@
+        """Starts atomic consume/produce messages thread
+
+        Args:
+            wait (bool, optional): Wait for consume thread to finish or not.
+            Defaults to True.
+        """
+        self.logger.info("Start of atomic consume/produce of messages")
+        self.atomic_thread = self.create_thread(
+            self._consume_atomic_produce, thread_name="stream_atomic_thread")
+        if wait:
+            self.atomic_thread.join()
+        return
+
+    def _consume_atomic_produce(self, logger: logging.Logger,
+                                topic: TopicStatus) -> TopicStatus:
+        def time_since_last_msg_ms() -> int:
+            diff_ms = datetime.now().timestamp() - topic.last_message_ts
+            return int(diff_ms * 1000)
+
+        def _elapsed_ms(ts):
+            return int((datetime.now().timestamp() - ts) * 1000)
+
+        def get_high_watermarks(topic_name):
+            partitions = [
+                ck.TopicPartition(topic_name, p)
+                for p in consumer.list_topics().topics[topic_name].partitions
+            ]
+            # committed = consumer.committed(partitions, timeout=10)
+            hwms = {}
+            for p in partitions:
+                # Get the partitions low and high watermark offsets.
+                (lo, hi) = consumer.get_watermark_offsets(p,
+                                                          timeout=10,
+                                                          cached=False)
+                hwms[p.partition] = hi
+            return hwms
+
+        def reached_end():
+            """Slow way to check if partition reached end offset
+            Whole check lasts ~900ms which is very slow for high scale version
+            of the workload
+            """
+            high_watermarks = get_high_watermarks(topic.source_topic_name)
+
+            assignments = {tp.partition for tp in consumer.assignment()}
+            if len(assignments) == 0:
+                return False
+
+            positions = consumer.position([
+                ck.TopicPartition(topic.source_topic_name, p)
+                for p in assignments
+            ])
+            end_for = {
+                p.partition
+                for p in positions if p.partition in high_watermarks
+                and high_watermarks[p.partition] <= p.offset
+            }
+
+            self.logger.debug(
+                f"[{topic.transaction_id}] Topic {topic.source_topic_name} "
+                f"partitions high watermarks {high_watermarks}, "
+                f"assignment: {assignments} positions: {positions}, "
+                f"end_for: {end_for}")
+
+            topic.partitions_eof |= end_for
+            consumers = self.consumer_count
+            if len(topic.partitions_eof) == len(high_watermarks):
+                return True
+            return len(end_for) == len(assignments) and consumers > 1
+
+        consumer = ck.DeserializingConsumer(topic.consumer_config)
```

Comment:
do we need complexity around deserializing the contents of the kvs? Just consume and produce to keep it simple?

Reply:
This was to follow the same path that other verifier does. We surely can remove it in future.

Reply:
Removed it in latest commit